source/module_esolver/esolver_fp.cpp

```diff
@@ -122,18 +123,20 @@ void ESolver_FP::after_scf(const int istep)
     if (istep % PARAM.inp.out_interval == 0)
     {
         // 3) write charge density
-        if (PARAM.inp.out_chg[0])
+        if (PARAM.inp.out_chg[0] > 0)
         {
             for (int is = 0; is < GlobalV::NSPIN; is++)
             {
                 double* data = nullptr;
                 if (PARAM.inp.dm_to_rho)
                 {
                     data = this->pelec->charge->rho[is];
+                    this->pw_rho->real2recip(this->pelec->charge->rho[is], this->pelec->charge->rhog[is]);
```

Comment:
```suggestion
                    this->pw_rhod->real2recip(this->pelec->charge->rho[is], this->pelec->charge->rhog[is]);
```
this->pw_rhod should be used here, which is the dense FFT grids in the case of uspp. All changes are needed in the PR.

Reply:
thanks, I will fix it later